src/mugs.js

```diff
@@ -295,7 +295,44 @@ define([
             rawControlXML: {
                 presence: 'optional',
                 lstring: 'Raw XML'
-            }
+            },
+            dataParent: {
+                lstring: 'Data Parent',
+                visibility: function(mug) {
+                    function recFunc(mug) {
+                        if (!mug) {
+                            return true;
+                        } else if (mug.__className === 'Repeat') {
+                            return false;
+                        }
+                        return recFunc(mug.form.tree.getNodeFromMug(mug).parent.value);
+                    }
+
+                    return recFunc(mug);
+                },
+                presence: 'optional',
+                widget: widgets.droppableText,
+                validationFunc: function(mug) {
+                    var dataParent = mug.p.dataParent,
+                        form = mug.form,
+                        dataParentMug;
+
+                    if (dataParent) {
+                        dataParentMug = form.getMugByPath(dataParent);
+
+                        if(!dataParentMug &&
+                           form.getBasePath().slice(0, -1) !== dataParent) {
+                            return "Must be valid path";
+                        } else if (dataParentMug && dataParentMug.__className !== 'Group') {
+                            return "Data must be a child of a group";
```

Comment:
Can't check `isNestableGroup` as repeat groups have that same attribute. Will create a new one called `possibleDataParent` though